plugins/scaffolder-backend/src/scaffolder/actions/builtin/fetch/template.ts

```diff
@@ -0,0 +1,239 @@
+      const templateDir = resolvePath(workDir, 'template');
+
+      const targetPath = ctx.input.targetPath ?? './';
+      const outputDir = resolveSafeChildPath(ctx.workspacePath, targetPath);
+
+      if (
+        ctx.input.copyWithoutRender &&
+        !Array.isArray(ctx.input.copyWithoutRender)
+      ) {
+        throw new InputError(
+          'Fetch action input copyWithoutRender must be an Array',
+        );
+      }
+
+      await fetchContents({
+        reader,
+        integrations,
+        baseUrl: ctx.baseUrl,
+        fetchUrl: ctx.input.url,
+        outputPath: templateDir,
+      });
+
+      ctx.logger.info('Listing files and directories in template');
+      const allEntriesInTemplate = await globby(`**/*`, {
+        cwd: templateDir,
+        dot: true,
+        onlyFiles: false,
+        markDirectories: true,
+      });
+
+      const nonTemplatedEntries = new Set(
+        (
+          await Promise.all(
+            (ctx.input.copyWithoutRender || []).map(pattern =>
+              globby(pattern, {
+                cwd: templateDir,
+                dot: true,
+                onlyFiles: false,
+                markDirectories: true,
+              }),
+            ),
+          )
+        ).flat(),
+      );
+
+      // Create a templater
+      const templater = nunjucks.configure({
+        ...(ctx.input.cookiecutterCompat
+          ? {}
+          : {
+              tags: {
+                // TODO(mtlewis/orkohunter): Document Why we are changing the literals? Not here, but on scaffolder docs. ADR?
+                variableStart: '${{',
+                variableEnd: '}}',
+              },
+            }),
+        // We don't want this builtin auto-escaping, since uses HTML escape sequences
+        // like `&quot;` - the correct way to escape strings in our case depends on
+        // the file type.
+        autoescape: false,
+      });
+
+      if (ctx.input.cookiecutterCompat) {
+        // The "jsonify" filter built into cookiecutter is common
+        // in fetch:cookiecutter templates, so when compat mode
+        // is enabled we alias the "dump" filter from nunjucks as
+        // jsonify. Dump accepts an optional `spaces` parameter
+        // which enables indented output, but when this parameter
+        // is not supplied it works identically to jsonify.
+        //
+        // cf. https://cookiecutter.readthedocs.io/en/latest/advanced/template_extensions.html?highlight=jsonify#jsonify-extension
+        // cf. https://mozilla.github.io/nunjucks/templating.html#dump
+        templater.addFilter('jsonify', templater.getFilter('dump'));
+      }
+
+      // Cookiecutter prefixes all parameters in templates with
+      // `cookiecutter.`. To replicate this, we wrap our parameters
+      // in an object with a `cookiecutter` property when compat
+      // mode is enabled.
+      const parameters = ctx.input.cookiecutterCompat
+        ? { cookiecutter: ctx.input.values }
```

Comment:
wondering if we still want to do something similar for the `fetch:template` action. Instead of them being on global scope, thinking about clashes with filter functions etc and extending the API at a later date. Maybe `${{ parameters.something }}`?

Reply:
Thanks for the suggestion Ben, seems reasonable to me - how about `values` for the name? It's a bit shorter, which seems like quite a nice property for a prefix that's going to show up all over the place in templates, and it also matches template.yaml.

FWIW it seems like values in the context don't inherently conflict with filters - adding a variable named `dump` worked fine, even though `dump` is also a valid filter name. It does seem worth namespacing variables from template.yaml though, perhaps we'll want to add some built in ones later.

Reply:
Nice! Done in 56b61c2bc.

Reply:
Sweet!